File: nim/n3xpl0rer.nim
import terminal, strutils
import modules/workflower
import modules/worker

proc showMenu() =
  while true:
    setForegroundColor(fgCyan)
    echo "============================="
    echo "          n3xpl0rer          "
    echo "============================="
    resetAttributes()
    echo "[1] Recon"
    echo "[2] Load a Workflow"
    echo "[3] Run a Workflow"
    echo "[4] Exit"
    setForegroundColor(fgCyan)
    echo "============================="
    resetAttributes()
    stdout.write("Select an option: ")
    let choice = readLine(stdin).strip()

    case choice
    of "1":
      echo "Recon: Not implemented"
    of "2":
      stdout.write("Enter the JSON file name in the 'workflows' folder: ")
      let filename = readLine(stdin).strip()
      let filepath = "../workflows/" & filename
      try:
        let workflow = workflower.loadWorkflow(filepath)
        workflower.displayWorkflow(workflow)
      except CatchableError as e:
        setForegroundColor(fgRed)
        echo "Error loading the workflow: ", e.msg
        resetAttributes()
      echo "Press Enter to continue..."
      discard readLine(stdin)
    of "3":
      stdout.write("Enter the JSON file name in the 'workflows' folder: ")
      let filename = readLine(stdin).strip()
      let filepath = "../workflows/" & filename
      stdout.write("Enter the target host: ")
      let host = readLine(stdin).strip()
      try:
        let workflow = workflower.loadWorkflow(filepath)
        worker.runWorkflow(workflow, host)
      except CatchableError as e:
        setForegroundColor(fgRed)
        echo "Error executing the workflow: ", e.msg
        resetAttributes()
      echo "Press Enter to continue..."
      discard readLine(stdin)
    of "4":
      setForegroundColor(fgYellow)
      echo "Exiting..."
      resetAttributes()
      quit(0)
    else:
      setForegroundColor(fgRed)
      echo "Invalid option. Please select a valid option."
      resetAttributes()
      echo "Press Enter to continue..."
      discard readLine(stdin)

when isMainModule:
  showMenu()

File: python/n3xpl0rer.py
from rich.console import Console
from rich.prompt import Prompt
from modules.workflower import Workflower
from modules.worker import Worker

class n3xpl0rer:
    def __init__(self):
        self.console = Console()
        self.menu_options = ["Display Workflow", "Run Workflow", "Exit"]

    def display_menu(self):
        ascii_art = r"""
[bold green]

👾 n̲3̲x̲p̲l̲0̲r̲e̲r̲ 👾

[/bold green]
        """
        self.console.print(ascii_art)

        for idx, option in enumerate(self.menu_options, start=1):
            self.console.print(f"[bold cyan]{idx}[/bold cyan]. {option}")
        self.console.print("\n" + "=" * 40)


    def run(self):
        while True:
            self.display_menu()
            choice = Prompt.ask("Select an option", choices=[str(i) for i in range(1, len(self.menu_options) + 1)])

            if choice == "1":
                self.display_workflow()
            elif choice == "2":
                self.run_workflow()
            elif choice == "3":
                self.console.print("Exiting...", style="bold red")
                break

    def display_workflow(self):
        workflow_path = input("Enter the path to the workflow JSON file: ")
        workflower = Workflower(workflow_path)
        workflower.display_workflow()

    def run_workflow(self):
      try:
        workflow_path = input("Enter the path to the workflow JSON file: ")
        workflower = Workflower(workflow_path)
        worker = Worker()
        workflow = workflower.load_workflow()
        self.console.print(f"Loaded workflow: {workflow['name']}", style="bold blue")
        worker.process_workflow(workflow)
      except FileNotFoundError as e:
        self.console.print(f"Workflow {workflow_path} not found", style="red")

if __name__ == "__main__":
    app = n3xpl0rer()
    app.run()
